packages/theme-base/dist/pages/_.vue

```diff
@@ -1,49 +1,32 @@
 <template>
-  <div :key="$route.path">
-    <component :is="getComponent" :cms-page="cmsPage" :page="page" />
+  <div :key="$route.path" style="min-height: 55vh">
+    <component
+      :is="getComponent"
+      :cms-page="cmsPage"
+      :page="page"
+      :error="error"
+    />
   </div>
 </template>
 <script>
 import { useCms, useNotifications } from "@shopware-pwa/composables";
+import { computed } from "vue-demi";
 
 export default {
   name: "DynamicRoute",
   components: {},
-  watchQuery(newQuery, oldQuery) {
-    // Only execute component methods if currency changed
-    return newQuery.currencyId !== oldQuery.currencyId;
-  },
-  asyncData: async ({ params, app, error: errorView, query, redirect }) => {
-    const { search, page, error } = useCms(app);
-    const { pushError } = useNotifications(app);
-    const searchResult = await search(params.pathMatch, query);
-
-    // direct user to the error page (keep http status code - so do not redirect)
-    if (error.value) {
-      if (error.value.statusCode === 404) {
-        errorView(error.value);
-      } else {
-        // show the error other than 404 in SwNotifications
-        pushError(error.value.message);
-      }
-    }
+  setup(props, { root }) {
+    const { page, error, loading } = useCms(root);
 
-    const unwrappedPage = page && page.value ? page.value : searchResult;
-    const name =
-      unwrappedPage && unwrappedPage.cmsPage && unwrappedPage.cmsPage.name;
-    const cmsPage = unwrappedPage && unwrappedPage.cmsPage;
-
-    // redirect to the cannnical URL if current path does not match the canonical one
-    if (params.pathMatch && unwrappedPage && unwrappedPage.canonicalPathInfo) {
-      if (unwrappedPage.canonicalPathInfo !== params.pathMatch) {
-        redirect(301, app.$routing.getUrl(unwrappedPage.canonicalPathInfo));
```

Comment:
shouldn't we move this logic to the page middleware as well?

Reply:
do you mean this?
```js
const cmsPage = computed(() => page.value?.cmsPage);
const cmsPageName = computed(() => cmsPage.value?.name);
```
Cms name is only for DevTools, I believe we should pass the whole CMS page response into specific view (like in product page) so this computed won't be necessary anymore :)

Reply:
good point, thanks!